runtime/src/main/java/org/corfudb/runtime/object/MVOCache.java

```diff
@@ -86,12 +81,19 @@ public void put(@Nonnull VersionedObjectIdentifier voId, @Nonnull T object) {
     }
 
     /**
-     * Register an MVO with this MVOCache.
-     * @param objectId The stream id of the provided MVO.
-     * @param mvo      The actual MVO being registered.
+     * Invalidate all versioned objects with the given object ID.
+     * @param objectId
      */
-    public void registerMVO(@Nonnull UUID objectId, @Nonnull MultiVersionObject<T> mvo) {
-        allMVOs.putIfAbsent(objectId, mvo);
+    public void invalidateAllVersionsOf(@Nonnull UUID objectId) {
+        if (log.isTraceEnabled()) {
+            log.trace("MVOCache: performing a invalidateAllVersionsOf for {}", objectId);
+        }
+
+        List<VersionedObjectIdentifier> voIdsToInvalidate =
+                objectCache.asMap().keySet().stream()
+                        .filter(voId -> objectId.equals(voId.getObjectId()))
+                        .collect(Collectors.toList());
+        objectCache.invalidateAll(voIdsToInvalidate);
```

Comment:
I think it would be worth having a test case that shows that this method behaves correctly even as other threads are populating the cache (and causing LRU evictions) with versions from different tables.

Reply:
That sounds like writing a test for Guava Cache? I think invalidating a key that does not exist, and calling `invalidate` and `put` in parallel should be already taken care by the Cache.

Reply:
Not so much as a test for the Guava cache, but more like a blackbox test for this API that validates the behaviour, irrespective of its implementation or of the implementation of the underlying cache. If we ever change the implementation or change the cache backing it, such a test can help us catch errors earlier on if any implicit assumptions being made no longer hold.